app/src/main/java/com/example/erabook/fragments/login/LoginFragment.kt

```diff
@@ -63,46 +72,59 @@ class LoginFragment : Fragment() {
-                                ).apply {
-                                    putExtra(DESTINATION_PROFILE, R.id.userProfileFragment)
-                                })
-                            requireContext().showToast(R.string.login_successful)
-                        } else {
-                            val exception = task.exception as FirebaseException
-                            exception.let {
-                                when (it.message) {
-                                    getString(R.string.error_code_message_incorrect) -> requireContext().showToast(
-                                        R.string.wrong_password_email
-                                    )
+                authenticationViewModel.signInWithEmailAndPassword(emailLogIn.trim(), passwordLogIn)
+                authenticationViewModel.isTaskSuccessful.observe(viewLifecycleOwner) { task ->
+                    if (task == true) {
+                        findNavController().navigate(R.id.loginToProfile)
+                        requireContext().showToast(R.string.login_successful)
+                    } else {
+                        authenticationViewModel.exception.let { message ->
+                            when (message.value) {
+                                getString(R.string.error_code_message_incorrect) -> requireContext().showToast(
+                                    R.string.wrong_password_email
+                                )
 
-                                    getString(R.string.error_code_message_connection) -> requireContext().showToast(
-                                        R.string.error_connection
-                                    )
-                                }
+                                getString(R.string.error_code_message_connection) -> requireContext().showToast(
+                                    R.string.error_connection
+                                )
                             }
                         }
                     }
+                }
             }
             forgotPasswordTextView.setOnClickListener {
-                authenticationViewModel.userLiveData.observe(viewLifecycleOwner) { user ->
-                    if (user != null) {
-                        authenticationViewModel.logout()
-                    } else {
-                        startActivity(
-                            Intent(
-                                requireContext(),
-                                ForgotPasswordActivity::class.java
-                            )
-                        )
+                startActivity(
+                    Intent(
+                        requireContext(),
+                        ForgotPasswordActivity::class.java
+                    )
+                )
+            }
+        }
+    }
+
+    @Deprecated("Deprecated in Java")
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
```

Comment:
onActivityResult is deprecated, please use the new way of listening for activity result: ActivityResultLauncher, do a research. Useful link [https://stackoverflow.com/questions/62671106/onactivityresult-method-is-deprecated-what-is-the-alternative](https://stackoverflow.com/questions/62671106/onactivityresult-method-is-deprecated-what-is-the-alternative)